src/decoder/stream.rs

```diff
@@ -244,6 +252,58 @@ impl JpegReader {
     }
 }
 
+#[repr(u8)]
+enum JpegTagApp14Transform {
+    // App14TransformYCCK = 2,
+    // App14TransformYCbCr = 1,
+    App14TransformUnknown = 0,
+}
+
+fn add_app14segment(jpeg_tables: &mut Vec<u8>, transform: JpegTagApp14Transform) {
```

Comment:
I'm not especially confident that this code is correct since I don't know much about JPEG encoding and we don't have any testcases that exercise it. Which is probably related to why the previous PR lingered. But it does fix decoding for some images. So, even though I'm not thrilled about it, I still think we should probably just merge this now.

Reply:
I guess one possibility would to alter `jpeg-decoder::Decoder` to allow overriding the `color_transform`? 

I can have a look at the `jpeg_tables` for the styles of data that I have, to see whether I can see where the colour is defined/why the original code treated it as YCrCb instead of RGB, but I think any change to try and resolve this in a more satisfactory manner would constitute a separate PR.